src/Language/Mulang/Transform/Normalizer.hs

```diff
@@ -118,8 +119,9 @@ normalizeReturn _   e             | isImplicitReturn e = Return e
 normalizeReturn _   (Sequence es) | Just (i, l) <- unwind es, isImplicitReturn l = Sequence $ i ++ [Return l]
 normalizeReturn _   e             = e
 
-normalizeTryCases    ops = map (\(p, e) -> (p, normalize ops e))
-normalizeSwitchCases ops = map (\(e1, e2) -> (normalize ops e1, normalize ops e2))
+normalizeRecordUpdates ops = map (\(i, e) -> (i, normalize ops e))
+normalizeTryCases      ops = map (\(p, e) -> (p, normalize ops e))
+normalizeSwitchCases   ops = map (\(e1, e2) -> (normalize ops e1, normalize ops e2))
```

Comment:
_Perhaps_ we could generalize those tree functions now. The first two are the same, and are mostly the same in all files.